### PR TITLE
Clean up FullCLR related scripts/docs

### PR DESCRIPTION
**Pull request by daxian-dbw**
Related to #4357 

This PR includes following changes:
1. Remove FulLCLR related build script in `build.psm1`
1. Remove commented out FullCLR related scripts in `AppVeyor.psm1`
1. Remove FullCLR test file `test/fullclr/PowerShellGithubDev.Tests.ps1`
1. Remove FullCLR related doc content.

### CODE REVIEW

File: build.psm1 [hunk 1]
```diff
@@ -907,12 +853,8 @@ function Start-PSPester {
     if ($Environment.IsWindows) {
         $Command += "Set-ExecutionPolicy -Scope Process Unrestricted; "
     }
-    $startParams = @{binDir=$binDir}
 
-    if(!$FullCLR)
-    {
-        $Command += "Import-Module '$Pester'; "
-    }
+    $Command += "Import-Module '$Pester'; "
```
Comment (adityapatwardhan) [inline build.psm1]: Do we need this, now that we have module autoloading?

**Review by adityapatwardhan — APPROVED** [review]
LGTM with minor comment

**Review by iSazonov — APPROVED** [review]

**Review by TravisEz13 — APPROVED** [review]
Doc changes look good

File: build.psm1 (hunk 1 above)
Comment (daxian-dbw) [inline build.psm1]: I think we should keep it as the `-Pester` parameter can be given a different path than the one in our module path.

**Pull request merged**